packages/aws-cdk-lib/aws-cloudwatch/lib/alarm.ts

```diff
@@ -208,26 +248,71 @@ export class Alarm extends AlarmBase {
       ...metricProps,
     });
 
+    this.onCfnAlarmCreated(alarm);
+
     this.alarmArn = this.getResourceArnAttribute(alarm.attrArn, {
       service: 'cloudwatch',
       resource: 'alarm',
       resourceName: this.physicalName,
       arnFormat: ArnFormat.COLON_RESOURCE_NAME,
     });
+
     this.alarmName = this.getResourceNameAttribute(alarm.ref);
 
     this.metric = props.metric;
-    const datapoints = props.datapointsToAlarm || props.evaluationPeriods;
-    this.annotation = {
-      // eslint-disable-next-line max-len
-      label: `${this.metric} ${OPERATOR_SYMBOLS[comparisonOperator]} ${props.threshold} for ${datapoints} datapoints within ${describePeriod(props.evaluationPeriods * metricPeriod(props.metric).toSeconds())}`,
-      value: props.threshold,
-    };
+    this.annotation = this.createHorizontalAnnotation(comparisonOperator, props);
 
     for (const [i, message] of Object.entries(this.metric.warningsV2 ?? {})) {
       Annotations.of(this).addWarningV2(i, message);
     }
   }
+  protected validateComparisonOperator(comparisonOperator: ComparisonOperator): void {
+    if (Alarm.isAnomalyDetectionOperator(comparisonOperator)) {
+      throw new ValidationError(
+        'Anomaly detection operators require to use the construct AnomalyDetectionAlarm.',
+        this,
+      );
+    }
+  }
```

Comment:
Yeah see my above comment. I don't think this is a good idea.

People should be allowed to use an `Alarm` to define an anomaly alarm (in fact, they probably already are today, because they have no other choice!)... it's just that the new construct makes it easier.